### ISSUE TITLE
How can one get image coordinates corresponding to an RoI during training?

### ISSUE THREAD

**Issue by SupreethN** [facebookresearch/detectron2#1402]
Suppose when training a Mask-RCNN I want to get image coordinates corresponding to features extracted by RoI align. 
For example, in box head we extract features using RoI align and perform computation over this RoI. I want to know the location (in the original image) of such a pooled feature. How can I do this?

**Comment by ppwwyyxx** [comment]
Detectron2 does not have a function that computes this, and the question is not related to detectron2 if you're looking for a formula to compute it. Not sure which one you're looking for, though.

**Comment by dhaivat666** [comment]
I have done something similar in detectron2. It is not that straightforward, you have to carefully return things but it can be done. I guess these are the places where you may need to make changes. 

1. Your features are computed in [this line](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/roi_heads/roi_heads.py#L649). And the corresponding bounding boxes are in `proposals` variables. 
2. If you can return both these things, you should get co-ordinates corresponding to your pooled features. However, the co-ordinates are still going to be in resized image space. 
3. To get co-ordinates in original image space, you would want to use the post-processing function of the forward pass. I used the one that is found [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/meta_arch/rcnn.py#L256). 

While you use this, be careful as to what all are you returning in these functions to ensure that you don't break the training procedure of inference process.

**Comment by SupreethN** [comment]
Just a clarification, suppose I have two models. 
One is a pretrained Mask-RCNN in eval mode that is used to extract bounding boxes for an image.
The second one is a Mask-RCNN I am using to train some task. 

Suppose I build training and testing dataloaders using the config file for second Mask-RCNN.
I pass the items in this dataloader to the first Mask-RCNN and get bounding boxes (without post-processing so they are not in original image space, directly from [here](https://github.com/facebookresearch/detectron2/blob/master/detectron2/modeling/roi_heads/roi_heads.py#L697).
Now, if I want to use these bounding boxes to extract some features in roi heads of the second Mask-RCNN, will they be in same image space or should I resize them?
